pinot-spi/src/main/java/org/apache/pinot/spi/utils/PinotReflectionUtils.java

```diff
@@ -48,6 +48,7 @@ public static Set<Class<?>> getClassesThroughReflection(String regexPattern, Cla
     return getClassesThroughReflection(PINOT_PACKAGE_NAME, regexPattern, annotation);
   }
 
+  @SuppressWarnings("deprecation")
```

Comment:
Are the changes in this file related?

Reply:
Ahh not really, but felt proabably better to add the suppress warnings for the previous https://github.com/apache/pinot/pull/12958 as you have suggested.

Reply:
Oh, I'd suggest suppressing just that one statement by adding `//noinspection deprecation` above it, then the comment and the suppress are next to each other. Can we keep it separate from this PR?

Reply:
Sure we can separate this